### PR TITLE
Remove domain/catch hook errors

### PR DESCRIPTION
**Pull request by NelsonVides**

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/139178) / small_tests / 1dc77033e4ba2030e0b6a55427c6b4facb4fe363
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3775/139178/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3775/139178/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-23_18.29.16/lib.mongooseim.logs/run.2022-09-23_18.29.16/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/139265) / small_tests / 5c606f7d15f6fa2dd6f8a9df27af4d9f0050355c
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3775/139265/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3775/139265/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-24_11.51.12/lib.mongooseim.logs/run.2022-09-24_11.51.12/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/139307) / small_tests / 67685246f4ca03e58f71c0603e92935176361bab
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3775/139307/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3775/139307/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-24_12.28.57/lib.mongooseim.logs/run.2022-09-24_12.28.57/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/139516) / small_tests / 0ee3dfbc2171c950d4e1ee8a0dcdeed21f08146d
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3775/139516/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3775/139516/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-26_18.45.59/lib.mongooseim.logs/run.2022-09-26_18.46.00/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/139570) / small_tests / f8754165ce781a21248d99b10cd6a8ec58171ad8
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3775/139570/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3775/139570/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-26_22.12.50/lib.mongooseim.logs/run.2022-09-26_22.12.50/suite.log.html)

File: src/mam/mod_mam_rdbms_arch.erl [hunk 1]
```diff
@@ -341,15 +341,18 @@ remove_archive(Acc, HostType, ArcID, _ArcJID) ->
...
+-spec remove_domain(mongoose_domain_api:remove_domain_acc(), host_type(), jid:lserver()) ->
+    mongoose_domain_api:remove_domain_acc().
 remove_domain(Acc, HostType, Domain) ->
-    {atomic, _} = mongoose_rdbms:sql_transaction(HostType, fun() ->
-            mongoose_rdbms:execute_successfully(HostType, mam_remove_domain, [Domain]),
-            mongoose_rdbms:execute_successfully(HostType, mam_remove_domain_prefs, [Domain]),
-            mongoose_rdbms:execute_successfully(HostType, mam_remove_domain_users, [Domain])
-        end),
-    Acc.
+    F = fun() ->
```
Comment (chrzaszcz) [inline src/mam/mod_mam_rdbms_arch.erl:347]: It is really unexpected that this function is (and most likely was already) untouched by tests. Could you investigate? I thought we had tests for domain removal for MAM.

**Review by chrzaszcz — COMMENTED** [review]
Looks good in general, I added minor comments.

File: src/mam/mod_mam_rdbms_arch.erl (hunk 1 above)
Comment (NelsonVides) [reply src/mam/mod_mam_rdbms_arch.erl:347]: No idea, I just ran locally with coverage and it is green in the report generated here, I don't know why it wasn't for codecov 😕

Comment (NelsonVides) [reply src/mam/mod_mam_rdbms_arch.erl:347]: codecov says that for master is also red: https://app.codecov.io/gh/esl/MongooseIM/blob/master/src/mam/mod_mam_rdbms_arch.erl so well, I'm sure we have a test for it, in https://github.com/esl/MongooseIM/blob/e79f08474226283c5de226ce7aeea2734a8112f1/big_tests/tests/domain_removal_SUITE.erl#L167, so it seems like something's off with codecov 😅

Comment (chrzaszcz) [reply src/mam/mod_mam_rdbms_arch.erl:347]: Ok, let's leave it for now

**Review by chrzaszcz — APPROVED** [review]

**Pull request merged**